terraform/versions.tf

```diff
@@ -0,0 +1,9 @@
+terraform {
+  required_version = ">= 1.5"
+  required_providers {
+    juju = {
+      source  = "juju/juju"
+      version = "~> 0.14.0"
```

Comment:
Not sure what we should standardise our .gitignores to for TF but I found this from one of our spec examples:

https://github.com/canonical/grafana-k8s-operator/pull/350/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947

There are a lot more TF files created than just in `.terraform/`